include/ocpp/v201/smart_charging.hpp

```diff
@@ -106,6 +148,16 @@ class SmartChargingHandler : public SmartChargingHandlerInterface {
     ///
     std::vector<ChargingProfile> get_profiles() const;
 
+    ///
+    /// \brief Clears profiles from the system using the given \p request
+    ///
+    ClearChargingProfileResponse clear_profiles(const ClearChargingProfileRequest& request) override;
+
+    ///
+    /// \brief Gets the charging profiles for the given \p request
+    ///
+    std::vector<ReportedChargingProfile> get_profiles(const GetChargingProfilesRequest& request) const override;
```

Comment:
I would name this differently, as we've got `get_profiles()` above that does something completely differently. When reading through the tests, I got confused about what we were doing when we called `get_profiles()`. One name would be `get_reported_profiles()` or `get_matching_profiles()`.

Reply:
I wouldnt say that the two functions do something completely different. But given that the return type is different I agree that `get_reported_profiles` reads better.